### ISSUE TITLE
How to invoke OkHttpClient.Builder's sslSocketFactory() in Mapbox v10 

### ISSUE THREAD

**Issue by maxtracking** [mapbox/mapbox-maps-android#517]
In my existing code I needed to set few things on OkHttpClient like this:
```
fun createOkHttpClient(context: Context): OkHttpClient {
        val builder = OkHttpClient().newBuilder()
        builder.hostnameVerifier(HostnameVerifier { _, _ -> return@HostnameVerifier true }) // <-- for simplicity here
        builder.addInterceptor { chain ->
            chain.proceed(
                chain.request().newBuilder()
                    .addHeader("SomeHeader", value).build()
            )
        }
        val (socketFactory, trustManager) = SslProvider.getSslObjects(context)
        builder.sslSocketFactory(socketFactory, trustManager)
        return builder.build()
    }
```
..and then set it like
```
HttpRequestUtil.setOkHttpClient(createOkHttpClient(context))
```

How can I do the same in v10 without providing a complete implementation of `HttpServiceInterface` ? From what I could see so far, `MapboxOkHttpService` doesn't expose anything to set/update its `httpClient`. Any example is appreciated.

**Comment by VincentJoshuaET** [comment]
Maybe this will help? https://docs.mapbox.com/android/beta/maps/guides/migrate-to-v10/#modular-architecture

**Comment by maxtracking** [comment]
@VincentJoshuaET  Thanks for the suggestion, it is exactly the subject of my question :) 
> "How can I do the same in v10 **without** providing a complete implementation of HttpServiceInterface"

In the previous versions, one didn't need to implement the whole class but just build and configure existing code. I can of course look into `MapboxOkHttpService` and implement all necessary method etc etc but I hoped that maybe I missed something and it's possible to avoid it.

**Comment by tobrun** [comment]
We have been working on a `HttpInterceptor` interface upstream. 
```java
/** Interface for intercepting and modifying HttpService function calls. */
public abstract class HttpServiceInterceptorInterface {
    protected long peer = 0;
    protected native void finalize() throws Throwable;
    /**
     * The function to intercept HTTP requests with.
     * @param request The original HttpRequest object to be modified.
     *
     * @return The modified HttpRequest object.
     */
    @NonNull
    public abstract HttpRequest onRequest(@NonNull HttpRequest request);
    /**
     * The function to intercept HTTP download options with.
     * @param download The original DownloadOptions object to be modified.
     *
     * @return The modified DownloadOptions object.
     */
    @NonNull
    public abstract DownloadOptions onDownload(@NonNull DownloadOptions download);
    /**
     * The function to intercept HTTP responses with.
     * @param response The original HttpResponse object to be modified.
     *
     * @return The modified HttpResponse object.
     */
    @NonNull
    public abstract HttpResponse onResponse(@NonNull HttpResponse response);
}
```

This will allow you to `setHeaders` for each outgoing `HttpRequest`.
It's found in v18.0.0 of `com.mapbox.common:common`. 
This version is not released yet as part of maps SDK but it is the version currently found on `main`. 
Will be released with `android-v10.0.0-rc.8`
